src/Options.jl

```diff
@@ -723,14 +736,23 @@ $(OPTION_DESCRIPTIONS)
 
     @assert print_precision > 0
 
+    _autodiff_backend = if autodiff_backend isa Union{Nothing,AbstractADType}
+        autodiff_backend
+    else
+        ADTypes.Auto(autodiff_backend)
```

Comment:
@gdalle here's the usecase^ thanks again :+1:

Reply:
awesome! do you want me to give a quick review?

Reply:
It’s a pretty minor part of the PR so not needed yet. But once I try to get more backends working in the future it would be much appreciated, thanks! :)

Reply:
alright, don't hesitate to ping me as needed!